core/src/main/c/share/dedup.cpp

```diff
@@ -136,6 +143,7 @@ int64_t merge_dedup_long_index_int_keys(
     return dest - dest_index;
 }
 
+template<typename index_t>
```

Comment:
Rename to an uppercase as is standard convention in C++ to distinguish template params from actual types.

E.g. `IndexT`.